sql/structured.go

```diff
@@ -51,6 +52,35 @@ func validateName(name, typ string) error {
 	return nil
 }
 
+// allocateName sets desc.Name to a value that is not equalName to any
+// of tableDesc's indexes. allocateName roughly follows PostgreSQL's
+// convention for automatically-named indexes.
+func (desc *IndexDescriptor) allocateName(tableDesc *TableDescriptor) {
+	segments := make([]string, 0, len(desc.ColumnIDs)+2)
+	segments = append(segments, tableDesc.Name)
```

Comment:
I wonder if we need to include the table name in the index name. I realize this is following postgres, but it seems like stuttering. In your examples below, you have `c.c_foo_idx`.